pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -253,6 +254,10 @@ func manageOpenShiftGlobalCAConfigMap_v311_00_to_latest(kubeClient kubernetes.In
 	existing, err := client.ConfigMaps(util.TargetNamespace).Get(context.TODO(), "openshift-global-ca", metav1.GetOptions{})
 	// Ensure we create the ConfigMap for the global CA, and that it has the right labels
 	// Lifted from library-go for the most part
+
+	// Bug 1826183: The global CA bundle is not intended to be used with workloads which run `update-ca-trust extract` on their own.
+	// In 4.6 we need to create this ConfigMap with the trust bundle injected so that the controller manager does not have an outage during upgrade.
+	// In 4.7 and beyond this ConfigMap should be deleted.
```

Comment:
So to connect some dots:
- builds, either docker or s2i strategy, *always* call `update-ca-trust extract` now?
- if yes, then some clarification here (vs. explanations I assume exist in openshift/builder or openshift/openshift-controller-manager) on how the global CAs are picked up, so things work with the proxy in addition to the direct contact to external repos like nexus, artifactory, etc. would be good
- if no, then don't we still need global CA bundle for the "non update-ca-trust" scenarios

Reply:
>     * builds, either docker or s2i strategy, _always_ call `update-ca-trust extract` now?

Yes - this was added in https://github.com/openshift/builder/pull/158

>     * if yes, then some clarification here (vs. explanations I assume exist in openshift/builder or openshift/openshift-controller-manager) on how the global CAs are picked up, so things work with the proxy in addition to the direct contact to external repos like nexus, artifactory, etc. would be good

The new flow will be a follows:

1. ocm-o copies the user trusted CA to `openshift-controller-manager/openshift-user-ca`
2. For each build, openshift-controller-manager reads `openshift-user-ca` and copies that into a `ConfigMap` in the build's namespace, mounting it at `/var/run/configs/openshift.io/pki/tls-ca-bundle.pem` inside the build pod containers.
3. When builds run, it copies this CA to `/etc/pki/ca-trust/source/anchors` and runs `update-ca-trust extract`. This merges the custom CA with the default CAs provided by the builder image.

I will update with additional comments.